packages/aws-cdk/lib/api/util/sdk.ts

```diff
@@ -105,6 +124,20 @@ export class SDK implements ISDK {
       });
     }
 
+    if (options.endpoints) {
+      try {
+        this.endpoints = JSON.parse(options.endpoints) || {};
+      } catch (_e) {
```

Comment:
I don't love exception-based control flow. Should be more intentional about what is valid - I'm afraid the fallback here is a little sloppy (it'll silently ignore a number of invalid cases, where it should have thrown instead).

Reply:
Hmm.. Would something like:

```
try {
    this.endpoints = JSON.parse(options.endpoints) || {};
} catch (_e) {}

this.endpoints = this.endpoints || options.endpoints.split(/,/g).reduce<ServiceEndpoints>((acc, entry) => {
    const [service, url] = entry.split('=');
    if (service && url) {
        acc[service] = url;
    }
    return acc;
}, {});
```

be better?
After all it would make it easier to add multiple try/catch cases in the future for other formats.